lib/components/widgets/loginScreen/login_screen.dart

```diff
@@ -1,17 +1,94 @@
+import 'dart:convert';
 import 'package:flutter/material.dart';
 import 'package:flutter/gestures.dart';
+import 'package:http/http.dart' as http;
 import 'package:classroom_app/components/styles/color/colors.dart';
-import 'package:classroom_app/components/widgets/CustomButton/custom_button.dart';
 import 'package:classroom_app/components/widgets/SignUpScreen/sign_up_screen.dart';
 
-class LoginScreen extends StatelessWidget {
+class LoginScreen extends StatefulWidget {
   const LoginScreen({super.key});
 
+  @override
+  State<LoginScreen> createState() => _LoginScreenState();
+}
+
+class _LoginScreenState extends State<LoginScreen> {
+  final _usernameController = TextEditingController();
+  final _passwordController = TextEditingController();
+  bool _isLoading = false;
+
+  Future<void> _login() async {
+    setState(() {
+      _isLoading = true;
+    });
+
+    try {
+      final response = await http.post(
+        Uri.parse('http://10.0.2.2:8080/login'),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: jsonEncode({
+          'username': _usernameController.text,
+          'password': _passwordController.text,
+        }),
+      );
+
+      if (response.statusCode == 200) {
+        final data = jsonDecode(response.body);
+
+        if (data['success']) {
+          _showDialog(
+            title: 'Login Successful',
+            message: data['message'] ?? 'Welcome!',
+          );
+        } else {
+          _showDialog(
+            title: 'Login Failed',
+            message: data['message'] ?? 'Invalid credentials.',
+          );
+        }
+      } else {
+        throw Exception('Failed to log in');
+      }
+    } catch (e) {
+      _showDialog(
+        title: 'Error',
+        message: 'An error occurred: $e',
+      );
+    } finally {
+      setState(() {
+        _isLoading = false;
+      });
+    }
+  }
+
+  void _showDialog({required String title, required String message}) {
+    showDialog(
+      context: context,
+      builder: (context) {
+        return AlertDialog(
+          title: Text(title),
+          content: Text(message),
+          actions: [
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+              child: const Text('OK'),
+            ),
+          ],
+        );
+      },
+    );
+  }
```

Comment:
The _login method handles both the API call and the UI logic for displaying dialogs. It's better to separate the concerns by moving the API call logic to a separate method or service.

```
Future<void> _login() async {
  setState(() {
    _isLoading = true;
  });

  try {
    final response = await _performLogin();
    _handleResponse(response);
  } catch (e) {
    _showDialog(
      title: 'Error',
      message: 'An error occurred: $e',
    );
  } finally {
    setState(() {
      _isLoading = false;
    });
  }
}

Future<http.Response> _performLogin() {
  return http.post(
    Uri.parse('http://10.0.2.2:8080/login'),
    headers: {
      'Content-Type': 'application/json',
    },
    body: jsonEncode({
      'username': _usernameController.text,
      'password': _passwordController.text,
    }),
  );
}

void _handleResponse(http.Response response) {
  if (response.statusCode == 200) {
    final data = jsonDecode(response.body);

    if (data['success']) {
      _showDialog(
        title: 'Login Successful',
        message: data['message'] ?? 'Welcome!',
      );
    } else {
      _showDialog(
        title: 'Login Failed',
        message: data['message'] ?? 'Invalid credentials.',
      );
    }
  } else {
    throw Exception('Failed to log in');
  }
}
```